knoten/csm.py

```diff
@@ -139,6 +141,34 @@ def generate_latlon_boundary(camera, boundary, radii=None):
     alts : list
            List of altitude values
     '''
+    def generate_ground_point(image_pt, camera, dem, height, max_its, tolerance=0.0001):
+        # Assume some path to a dem has been provided
+        if not isinstance(image_pt, csmapi.ImageCoord):
+            # Support a call where image_pt is in the form (x,y)
+            image_pt = csmapi.ImageCoord(*image_pt)
+
+        if dem != '':
+            gd = GeoDataset(dem)
+            intersection = camera.imageToGround(image_pt, 0.0)
+            iterations = 0
+            while iterations != max_its:
+                lon, lat, alt = pyproj.transform(ecef, lla, intersection.x, intersection.y, intersection.z)
```

Comment:
You should define this outside of the while loop so it's not constantly recreated.